src/components/UserSelector.tsx

```diff
@@ -2,39 +1,97 @@
+import { useEffect, useRef, useState } from 'react';
+import { User } from '../types/User';
+import classNames from 'classnames';
+
+type Props = {
+  users: User[];
+  setUserSelected: (user: User) => void;
+  userSelected: User | null;
+  handleUserPosts: (user: User) => void;
+  setShowSideBar: (item: number | null) => void;
+};
+
+export const UserSelector: React.FC<Props> = ({
+  users,
+  setUserSelected,
+  userSelected,
+  handleUserPosts,
+  setShowSideBar,
+}) => {
```

Comment:
You have used `setUserSelected` and `setShowUsers` in the same function call. According to the checklist, it's better to avoid using `setState` several times in one function call for clarity of the code.